src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md

```diff
@@ -0,0 +1,266 @@
+In conclusion, this experiment proves that increasing osd_max_scrubs to reduce scrub duration is a viable option but the user should be careful to not increase the concurrent scrubs to a very high number since this affects the client throughput and latency.
+
+##### Reducing scrub cost
+
+Every operation has a cost associated with it. As discussed in the beginning of this section, the lower the cost of the operation the higher are the chances of it getting dequeued faster. The scrub operations, by default, have a cost of 50MB.
+WPQ uses a combination of priority and cost to decide which operations are dequeued. Due to this it is easy to control operations such as scrub. For example, the osd_requested_scrub_priority is 120, this means that when a user explicitly asks for a scrub instead of waiting for the scheduled scrub WPQ is able to prioritize scrub operations over client operations (priority of 63). We are looking at the cost parameter with the hope that it could do the same for mClock.
+
+For all the mClock tests mentioned below we have reduced the scrub cost to 20MB and for the WPQ tests we have used the default osd_requested_scrub_priority of 120.
+
+![](images/image12_scrub_time_with_modified_cost.png)
+
+In the above graph we can see the decrease in scrub duration for the mClock profiles. The scrub performance for the profiles looks similar to that of WPQ (with scrub priority = 120).
+We can see the same in the graphs comparing the average client latency and scrub rate over the test duration where the trends for high recovery profile and WPQ look similar.
+
+![](images/image13_avg_clat_WPQ_vs_highrecovery_modified_cost.png)
+
+![](images/image14_scrub_rate_WPQ_vs_high_recovery_modified_cost.png)
+
+While we see a better scrub performance by mClock profiles, let’s take a look at how the average throughput changes in comparison to WPQ. In the graph below, it is evident that the mClock profiles have better throughput than WPQ. In particular, the high client profile continues to give higher throughput than the other profiles and WPQ.
+Decreasing the osd scrub cost has helped the mClock profiles provide scrub performance that is comparable to WPQ’s user requested scrub. And without any loss in client throughput and latency!
+
+![](images/image15_IOPS_with_modified_cost.png)
+
+## Client Ops with Background Scrub & Recovery
+
+### Test Steps
+
+1. Bring up the Ceph cluster with 4 OSDs
+2. Create a client RBD pool with replication factor of 3 and prefill some objects into it.
+3. Create a RBD recovery pool with replication factor 3.
+4. Mark an OSD down and out. After the cluster handles the OSD down event, 50K 4MiB objects are pre-filled into the recovery pool.
```

Comment:
This made me think of the Simon and Garfunkel song "Bridge over Troubled Water" -- the lyrics go:
"When you're down and out
When you're on the street..."

;)